examples/hotrod/pkg/pool/pool.go

```diff
@@ -49,5 +49,7 @@ func (p *Pool) Execute(job func()) {
 
 // Stop halts all the workers
 func (p *Pool) Stop() {
-	p.stop <- struct{}{}
+	if p.stop != nil {
```

Comment:
why is the nil check necessary?

Reply:
@yurishkuro sorry for the late reply,
IMHO, this is a bug. Assuming you have more than 1 worker, calling stop only stops 1 of the workers and not all, while as closing the chan stops all the workers. Please let me know if you still want me to rename the PR.

Reply:
good point